components/calendar/src/any_calendar.rs

```diff
@@ -536,7 +577,10 @@ impl AnyCalendarKind {
         } else if *x == value!("ethioaa") {
             AnyCalendarKind::Ethioaa
         } else {
-            return None;
+            let mut string = x.to_string();
+            string.truncate(16);
+            let tiny = string.parse().unwrap_or(tinystr!(16, "unknown"));
```

Comment:
Nit (optional): use write_to_string and take a slice of it

Reply:
Won't that run the same code?

Reply:
(This is obsoleted by my new PR)